mitosheet/mitosheet/public/v3/sheet_functions/misc_functions.py

```diff
@@ -149,11 +150,59 @@ def GETNEXTVALUE(series: pd.Series, condition: BoolRestrictedInputType) -> pd.Se
 
     return GETPREVIOUSVALUE(reversed_series, reversed_condition)[::-1]
 
+@cast_values_in_arg_to_type('index', 'int')
+def VLOOKUP(lookup_value: AnyPrimitiveOrSeriesInputType, where: pd.DataFrame, index: IntRestrictedInputType) -> pd.Series:
+    """
+    {
+        "function": "VLOOKUP",
+        "description": "Returns the next value from series that meets the condition.",
+        "search_terms": ["vlookup", "merge", "join", "search", "lookup"],
+        "examples": [
+            "VLOOKUP(Names0, Sample_Names:Sample_Ages, 1, False)",
+            "VLOOKUP("Sample", Sample_Names:Sample_Ages, 2, False)",
+            "VLOOKUP(Names0, Sample_Names:Sample_Ages, [1, 2], False)"
+        ],
+        "syntax": "VLOOKUP(lookup_value, where, index, approximate)",
+        "syntax_elements": [{
+                "element": "lookup_value",
+                "description": "The value to look up."
+            }, {
+                "element": "where",
+                "description": "The range to look up in."
+            }, {
+                "element": "index",
+                "description": "The column index to return."
+            }
+        ]
+    }
+    """
+    
+    # If the lookup value and index are both a primitive, we don't need to merge. 
+    if not isinstance(lookup_value, pd.Series) and isinstance(index, int):
+        return where.loc[where.iloc[:,0] == lookup_value].iloc[0, index-1]
+    
+    value = get_series_from_primitive_or_series(lookup_value, where.index)
+    value.name = 'lookup_value'
+    
+    # Then we want to do a merge on the column we're looking up from, and the df we're looking up in.
+    where_deduplicated = where.drop_duplicates(subset=where.iloc[:,0].name)
```

Comment:
might be helpful to save `where.iloc[:,0]` as `where_first_column` for readability - took me a bit to parse.